src/tests/PingSiteTest/PingSiteTest.ts

```
@@ -0,0 +1,41 @@
+/*
+ Copyright (c), CKSource Holding sp. z o.o. All rights reserved.
+ */
+
+import { ITest, TestResults } from '../Test';
+import { PingSiteTestDefinition } from './PingSiteTestDefinition';
+
+class PingSiteTest implements ITest {
+	public testName: string = 'ping';
+
+	public constructor(
+		public testDefinition: PingSiteTestDefinition
+	) {}
+
+	public async run(): Promise<TestResults> {
+		const httpResponse: Response = await fetch( this.testDefinition.url );
+		const statusCode: number = httpResponse.status;
+
+		if ( statusCode > 399 ) {
+			console.log(
+				this.testDefinition.formatErrorMessage( this.testName, 'Wrong Status Code received', `Received: ${ statusCode }` )
+			);
+
+			return { status: 1 };
```

WDYT about Throwing error instead of returning results approach? We use it in CS and it works well :) And there is less code.

---

Just `throw new Error('Wrong Status Code received. Received: ${ statusCode }')` and logging can be handled in `TestRunner`

---

Here is an example: https://github.com/cksource/cs/blob/master/cs-e2e-tests/src/runner.ts#L154-L191

---

We decided to cover it in a new PR as a follow-up.